diffkemp/snapshot.py

```diff
@@ -207,38 +178,37 @@ def _from_yaml(self, yaml_file):
 
         self.llvm_version = yaml_dict["llvm_version"]
 
-        llvm_finder_cls = None
-        llvm_finder_path = None
+        # Override default source finder when necessary
         if yaml_dict["llvm_source_finder"]["kind"] == "kernel_with_builder":
```

Comment:
Perhaps a comment on why it is necessary to override the `LlvmSourceFinder` and `SourceTree` would be useful here.